### PR TITLE
feat: make `"KECCAK256"` an alias for `"SHA3"` when deserializing geth `debug_traceBlockBy*` response body

### PR DESCRIPTION
**Pull request by DavidCai1993**
When deserializing responses from geth `debug_traceBlockByHash` and `debug_traceBlockByNumber` APIs, the name for the opcode  `0x20`  is `"KECCAK256"`(which is used in [go-ethereum](https://github.com/ethereum/go-ethereum/blob/master/core/vm/opcodes.go)), not `"SHA3"` (which is [used now in `./evm-types`](https://github.com/appliedzkp/zkevm-circuits/blob/main/eth-types/src/evm_types/opcode_ids.rs#L444)):
```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": [
        {
            "result": {
                 "gas": 248742,
                 "failed": false,
                 "structLogs": [
                     {
                      "pc": 189,
                      *"op": "KECCAK256",*
                      "gas": 178805,
                      "gasCost": 42,
                      "depth": 1,
                      "stack": [
                          "0x3635c9adc5dea00000",
                          "0x40",
                          "0x0"
                      ],
                      "memory": [
                          "000000000000000000000000b8f67472dcc25589672a61905f7fd63f09e5d470",
                          "0000000000000000000000000000000000000000000000000000000000000000",
                          "00000000000000000000000000000000000000000000000000000000000000a0",
                          "0000000000000000000000000000000000000000000000000000000000000000",
                          "00000000000000000000000000000000000000000000003635c9adc5dea00000",
                          "00000000000000000000000000000000000000000000003635c9adc5dea00000"
                      ]
                    }
                 ]
             }
         }    
     ]
}
```

The response key name is defined in [opcodes.go](https://github.com/ethereum/go-ethereum/blob/master/core/vm/opcodes.go) in go-ethereum:
```go
var opCodeToString = map[OpCode]string{
	// ...
	// 0x20 range - crypto.
	KECCAK256: "KECCAK256",
        // ...
}
```

So when we call [`BuilderClient.gen_inputs(block_num)`](https://github.com/appliedzkp/zkevm-circuits/blob/main/bus-mapping/src/circuit_input_builder.rs#L1645) (by using `prover_cmd`), if the response body including `"op": "KECCAK256"` like the example above, the process will panic and throw:

```
Error("data did not match any variant of untagged enum ResponseData", line: 1, column: 46675)
```

So this PR is to fix this panic by making `"KECCAK256"` an alias for `"SHA3"` (and with a test case), since they are the same opcode name for `0x20`.

### CODE REVIEW

**Comment by ChihChengLiang** [comment]
Hi @DavidCai1993, thank you for the contribution! The formating seems not passed in the CI, can we run `make fmt`?

File: eth-types/src/evm_types/opcode_ids.rs [hunk 1]
```diff
@@ -748,6 +748,7 @@ impl FromStr for OpcodeId {
             "REVERT" => OpcodeId::REVERT,
             "INVALID" => OpcodeId::INVALID(0xfe),
             "SHA3" => OpcodeId::SHA3,
+            "KECCAK256" => OpcodeId::SHA3,
```
[the remark this answers is not in the document]
Comment (DavidCai1993) [reply eth-types/src/evm_types/opcode_ids.rs:751]: OK, i'll fix it👍

**Comment by DavidCai1993** [comment]
@ChihChengLiang @HAOYUatHZ changed to `"SHA3" | "KECCAK256"` and run `cargo fmt --all` 👍

**Review by CPerezz — APPROVED** [review]
LGTM!! 

Thanks for your contribution @DavidCai1993 :partying_face:

**Review by ChihChengLiang — APPROVED** [review]
LGTM, thank you for the contribution!

**Pull request merged**